discovery/kubernetes/endpoints.go

```diff
@@ -131,6 +131,15 @@ func NewEndpoints(l log.Logger, eps cache.SharedIndexInformer, svc, pod, node ca
 	if err != nil {
 		level.Error(l).Log("msg", "Error adding services event handler.", "err", err)
 	}
+	_, err = e.podInf.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(_, o interface{}) {
```

Comment:
maybe we can compare with `old` and only enqueue in case the `.status.phase` to processing changes that will lead to no metadata changes from our side.

Will not be that future-proof but it should be more optimal